### PR TITLE
Fixed local logging and added call stack to errors

### PR DESCRIPTION
**Pull request by D1v38om83r**

### CODE REVIEW

File: internal/configuration/configuration.go [hunk 1]
```diff
@@ -97,18 +99,26 @@ var getEnvironmentConfiguration = func() Configuration {
...
+			panic(fmt.Sprintf("unable to deserialize configuration from environment\n%+v", err))
 		}
 	}
 	return configuration
 }
 
 var SerializeConfiguration = func(configuration *Configuration) ([]byte, error) {
-	return json.Marshal(configuration)
+	value, err := json.Marshal(configuration)
+	if err != nil {
```
Comment (sidiesen) [inline internal/configuration/configuration.go]: AddStackToError already handles the case where err == nil
You can simplify this to:
```
value, err := json.Marshal(configuration)
return value, errorhelper.AddStackToError(err)
```

File: internal/configuration/configuration.go [hunk 2]
```diff
@@ -97,18 +99,26 @@ var getEnvironmentConfiguration = func() Configuration {
...
+		return value, errorhelper.AddStackToError(err)
+	}
+	return value, err
 }
 
 var DeserializeConfiguration = func(data []byte, configuration *Configuration) error {
-	return json.Unmarshal(data, &configuration)
+	err := json.Unmarshal(data, &configuration)
+	if err != nil {
```
Comment (sidiesen) [inline internal/configuration/configuration.go]: same here, simplify to:

`return errorhelper.AddStackToError(json.Unmarshal(...))`

File: internal/jrds/client.go [hunk 3]
```diff
@@ -194,29 +197,30 @@ func (jrds *JrdsClient) issuePostRequest(url string, payload interface{}, out in
 		}
 	}
 
-	return err
+	return errors.WithStack(err)
```
Comment (sidiesen) [inline internal/jrds/client.go]: errorhelper.AddStackToError()?

File: internal/jrds/client.go [hunk 4]
```diff
@@ -194,29 +197,30 @@ func (jrds *JrdsClient) issuePostRequest(url string, payload interface{}, out in
...
 }
 
 func (jrds *JrdsClient) issueGetRequest(url string, out interface{}) error {
 	code, body, err := jrds.client.Get(url, jrds.getDefaultHeaders())
 
 	if err != nil {
-		return NewRequestError(fmt.Sprintf("request error %v : %v\n", url, code))
+		return NewRequestError(fmt.Sprintf("request error %v : %v\n%+v", url, code, err))
```
Comment (sidiesen) [inline internal/jrds/client.go]: should this error string end with another '\n'?

File: internal/jrds/client.go [hunk 5]
```diff
@@ -194,29 +197,30 @@ func (jrds *JrdsClient) issuePostRequest(url string, payload interface{}, out in
...
 	}
 
 	if code == 401 {
 		return NewRequestAuthorizationError(fmt.Sprintf("authorization error %v : %v\n", url, code))
 	}
 
 	if code != 200 {
-		return NewRequestInvalidStatusError(fmt.Sprintf("invalid return code for %v : %v\n", url, code))
+		return NewRequestInvalidStatusError(
+			errorhelper.NewErrorWithStack(fmt.Sprintf("invalid return code for %v : %v\n", url, code)).Error())
```
Comment (sidiesen) [inline internal/jrds/client.go]: is the addition of .Error() at the end here on purpose?

File: internal/jrds/client.go (hunk 3 above)
Comment (D1v38om83r) [reply internal/jrds/client.go]: changed it to return err

errors.WithStack() is a part of pkg/errors which is a 3rd party package

File: internal/jrds/client.go (hunk 4 above)
Comment (D1v38om83r) [reply internal/jrds/client.go]: its not necessary to end the error message with another newline. %+v prints detailed errors.

File: internal/jrds/client.go (hunk 5 above)
Comment (D1v38om83r) [reply internal/jrds/client.go]: NewRequestInvalidStatusError() takes a string as argument. err.Error() returns the string.

**Pull request merged**